workflow/yadage/workflow.yaml

```diff
@@ -0,0 +1,79 @@
+# Note that if you are working on the analysis development locally, i.e. outside
+# of the REANA platform, you can proceed as follows:
+#
+#   $ cd reana-demo-lhcb-d2pimumu
+#   $ mkdir -p yadage-local-run/yadage-inputs
+#   $ cd yadage-local-run
+#   $ cp -a ../code yadage-inputs
+#   $ yadage-run . ../workflow/yadage/workflow.yaml \
```

Comment:
This `yadage-run` command does not work... 

```
   ...
    return combine_outputs(map(lambda s: select_reference(s, selection), steps), flatten, unwrapsingle)
  File "/home/simko/.virtualenvs/yadage/lib/python3.7/site-packages/yadage/handlers/expression_handlers.py", line 36, in select_reference
    raise RuntimeError('no matches found in reference selection. selection %s | result %s' % (selection, step['result']))
RuntimeError: no matches found in reference selection. selection datafile | result <TypedLeafs: {'data': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/data/D2PiMuMuOS.root', 'massfit': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/code/OSMassFit.C', 'modelfixing': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/code/ModelFixing.C', 'optimise': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/code/Optimise.C'}>
2019-09-09 17:12:27,756 |  yadage.steering_api |   INFO | done. dumping workflow to disk.
Error: Workflow failed: no matches found in reference selection. selection datafile | result <TypedLeafs: {'data': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/data/D2PiMuMuOS.root', 'massfit': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/code/OSMassFit.C', 'modelfixing': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/code/ModelFixing.C', 'optimise': '/home/simko/private/project/reana/src/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs/code/Optimise.C'}>
```

Reply:
Ammended, but I have observed that it works locally only the first time. Subsequent runs require the extra flag `--accept-metadir` that uses cache, but raises some errors with the docker image:
```console
2019-09-10 10:42:31,874 | pack.optimisation.st |  ERROR | non-zero return code raising exception
2019-09-10 10:42:31,874 | pack.optimisation.st |  ERROR | subprocess failed. code: 125,  command docker run --rm -i  --cidfile /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation/_packtivity/optimisation.cid     -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:rw -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:rw reanahub/reana-env-root6:latest sh -c sh
Traceback (most recent call last):
  File "/home/dprelipc/.virtualenvs/reana/lib/python3.6/site-packages/packtivity/handlers/execution_handlers.py", line 331, in execute_and_tail_subprocess
    returncode=proc.returncode, cmd=command_string
subprocess.CalledProcessError: Command 'docker run --rm -i  --cidfile /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation/_packtivity/optimisation.cid     -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:rw -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:rw reanahub/reana-env-root6:latest sh -c sh' returned non-zero exit status 125.
2019-09-10 10:42:31,875 | pack.optimisation.st |  ERROR | job execution if job {'name': 'optimisation', 'wflow_node_id': '507398e5-d731-4bf7-bbb1-1c5d2ee348bc', 'wflow_offset': '', 'wflow_stage': 'optimisation', 'wflow_stage_node_idx': 0, 'wflow_hints': {'is_purepub': False}} raise exception exception
Traceback (most recent call last):
  File "/home/dprelipc/.virtualenvs/reana/lib/python3.6/site-packages/packtivity/handlers/execution_handlers.py", line 331, in execute_and_tail_subprocess
    returncode=proc.returncode, cmd=command_string
subprocess.CalledProcessError: Command 'docker run --rm -i  --cidfile /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation/_packtivity/optimisation.cid     -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:rw -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:rw reanahub/reana-env-root6:latest sh -c sh' returned non-zero exit status 125.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dprelipc/.virtualenvs/reana/lib/python3.6/site-packages/packtivity/syncbackends.py", line 187, in run_packtivity
    run_in_env(job, env, state, metadata, pack_config, exec_config)
  File "/home/dprelipc/.virtualenvs/reana/lib/python3.6/site-packages/packtivity/syncbackends.py", line 125, in run_in_env
    return handler(exec_config, environment, state, job, metadata)
  File "/home/dprelipc/.virtualenvs/reana/lib/python3.6/site-packages/packtivity/handlers/execution_handlers.py", line 448, in docker_enc_handler
    result = run(config, state, log, metadata, rspec)
  File "/home/dprelipc/.virtualenvs/reana/lib/python3.6/site-packages/packtivity/handlers/execution_handlers.py", line 399, in run_containers_in_docker_runtime
    logging_topic="run",
  File "/home/dprelipc/.virtualenvs/reana/lib/python3.6/site-packages/packtivity/handlers/execution_handlers.py", line 338, in execute_and_tail_subprocess
    raise RuntimeError("failed container execution subprocess. %s", command_string)
RuntimeError: ('failed container execution subprocess. %s', 'docker run --rm -i  --cidfile /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation/_packtivity/optimisation.cid     -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/optimisation:rw -v /home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:/home/dprelipc/project/reana/reana-demo-lhcb-d2pimumu/yadage-local-run/yadage-inputs:rw reanahub/reana-env-root6:latest sh -c sh')
```

Hence, the `yadage-local-run` should be cleared before every execution.